### DISCUSSION TITLE
Long delay of 3-4s for each call

### DISCUSSION THREAD

**Discussion by Sascha-L** [alandtse/alexa_media_player#2549] [General]
Hi,

no matter if I call this:

`action: media_player.play_media
data:
  media_content_id: My Routine # Routine name in Alexa app, this is set to play on All Speakers
  media_content_type: routine
target:
  entity_id: media_player.mydevice # need to select a default player
`

or

`data:
      media_content_id: shuffle Playlist MyPlaylistName
      media_content_type: APPLE_MUSIC`

It always takes at least 3-4s until the echo device reacts on this command and then of course it also takes a few seconds until the echo device plays the music. For the second part it is up to Alexa of course, but how to improve the long delay after sending a command to the echo device and until it receives the command?

**Comment by danielbrunt57** [comment]
I'm sure log information is going to point to Amazon being at fault by restricting/throttling requests to the undocumented API thus forcing exponential backoff in retrying GET/POST requests.
I'm now seeing exponential backoff delays in excess of 15 seconds to retrieve history records to permit ascertaing the last_called device. But this varies depending on the time of day. At 1:30 am I'm seeing a total delay of 6 seconds to achieve a  200 response OK.  Other times can be less, or worse. 
We unfortunately have nothing we can do about it.
